src/viam/sdk/services/private/mlmodel_server.cpp

```diff
@@ -58,7 +53,35 @@ ::grpc::Status MLModelServiceServer::Infer(
                         << static_cast<ut>(tensor_type);
                 return helper.fail(::grpc::INVALID_ARGUMENT, message.str().c_str());
             }
-            inputs.emplace(std::move(input.name), std::move(tensor));
+            inputs.emplace(tensor_pair.first, std::move(tensor));
+        } else {
+            // Normal case: multiple tensors, do metadata checks
+	    // If there are extra tensors in the inputs that not found in the metadata,
+	    // they will not be passed on to the implementation.
```

Comment:
Added a comment to say that if the user provides more inputs than are necessary, the extra inputs will be skipped over